xc/common/utils/prjxray_create_pinmap_csv.py

```diff
@@ -66,13 +72,32 @@ def main():
 
     args = parser.parse_args()
 
+    db = prjxray.db.Database(args.db_root, args.part)
+    grid = db.grid()
+
+    CAPACITY_IOS = ["IPAD", "OPAD"]
+
     pin_to_iob = {}
     with sqlite3.connect(args.connection_database) as conn:
         for line in csv.DictReader(args.package_pins):
             assert line['pin'] not in pin_to_iob
-            loc = get_vpr_coords_from_site_name(conn, line['site'])
-            if loc:
-                pin_to_iob[line['pin']] = (line['site'], loc)
+            site = line['site']
+            loc = get_vpr_coords_from_site_name(conn, site)
+
+            if loc is None:
+                continue
+
+            gridinfo = grid.gridinfo_at_tilename(line['tile'])
+            sites = sorted(gridinfo.sites.keys())
```

Comment:
You should be able to get this from the connection_database?

Reply:
Done